libraries/AP_RangeFinder/AP_RangeFinder_Benewake_TFMiniPlus.cpp

```diff
@@ -132,6 +135,11 @@ bool AP_RangeFinder_Benewake_TFMiniPlus::init()
 fail:
     _dev->get_semaphore()->give();
     return false;
+
+    if (ver[0] * 10000 + ver[1] * 100 + ver[3] < 20003) {
+        gcs().send_text(MAV_SEVERITY_ERROR, "TFMini: FW ver %u.%u.%u (need>=2.0.3)",
+                            (unsigned)ver[0],(unsigned)ver[1],(unsigned)ver[2]);
+    }   
```

Comment:
this code will never execute since you have the return false just above this

Reply:
and you are checking the version again. I disagree with @peterbarker here: the failure is just on init and the additional time we hold the semaphore for this error condition doesn't matter much. I'd say to either move it where it was or change it in such a way you don't check the version twice.